worker.go

```diff
@@ -282,6 +290,49 @@ func (w *Worker) MetricsProcessedCount() int64 {
 	return w.processed
 }
 
+// SampleTimeseries takes a metric and counts whether the timeseries
```

Comment:
Can we add a benchmark, both for this function (`SampleTimeseries`) and for the `Work` function as a whole?